HTML `<search>` element (#27727)
mdn/content: files/en-us/web/html/element/search/index.md

## Attributes

This element only includes the [global attributes](/en-US/docs/Web/HTML/Global_attributes).

## Usage notes

It's not appropriate to use the `<search>` element just for presenting search results, though suggestions and links as part of "quick search" results can be included as part of a search feature. Rather, a returned web page of search results would instead be expected to be presented as part of the main content of that web page.

## Examples

### Header search form

Include a search form within a website header to perform a simple site-wide search.

```html
<header>
  <h1>Movies website</h1>
  <search>
    <form action="./search/">
      <label for="movie">Find a Movie</label>
      <input type="search" id="movie" name="q">
      <button type="submit">Search</button>
    </form>
  </search>
</header>
```

### Result

{{EmbedLiveSample('Header search form')}}

### Web app search

> JavaScript search functionality in a web application. No form element to perform a server-side submission, but for semantics a search element represents search and filtering capabilities.

Reviewer: ```suggestion
This example demonstrates potential DOM content when dynamically including JavaScript search functionality in a web application. When search functionality is implemented entirely with JavaScript, if no form is submitted, neither a {{HTMLElement("form")}} element nor a submit {{HTMLElement("button")}} is required. For semantics, the `<search>` element is included to contain the search and filtering capabilities.
```

```html
<search>
  <label>
    Find and filter your query
    <input type="search" id="query">
  </label>
  <label>
    <input type="checkbox" id="exact-only">
    Exact matches only
  </label>

  <section>
    <h3>Results:</h3>
    <ul id="results">
      <!-- search result content -->
    </ul>
    <output id="no-results">
      <!-- no results content -->
    </output>
  </section>
</search>
```

> ### Result

Reviewer: ```suggestion
#### Result
```

> {{EmbedLiveSample('Web app search')}}

Reviewer: ```suggestion
{{EmbedLiveSample('Web app search')}}

> **Note:** Remember that some users don't have JavaScript, and none of your users have JavaScript running until the JavaScript is successfully downloaded, parsed, and executed, ensure your users can access the content of your site with JavaScript disabled.
```

### Multiple searches

> A page with two search features. The first is a global site search located on the header. The second is a search and filter based on the page context, in our example a car search.

Reviewer: ```suggestion
This example demonstrates a page with two search features. The first is a global site search located on the header. The second is a search and filter based on the page context, in our example a car search.
```

```html
<body>
  <header>
    <h1>Car rental agency</h1>
    <search title="Website">
      ...
    </search>
  </header>
  <main>
    <h2>Cars available for rent</h2>
     <search title="Cars">
       <h3>Filter results</h3>
       ...
     </search>
     <article>
      <!-- search result content -->
    </article>
  </main>
</body>
```

> ### Result

Reviewer: ```suggestion
#### Result
```
